code/game/machinery/engi_points_vendor.dm

```diff
@@ -0,0 +1,674 @@
+			if(glow.light_color == "nw")
+				glow.light_color = "green"
+				glow.light_range = glow.range * 1.1
+				glow.light_power = glow.light_power * 2 // Any changes to power must come in pairs to neutralize it for other colors
+				glow.update_light()
+				continue
+			if(glow.light_color == "green")
+				glow.light_color = "sw"
+				glow.light_power = glow.light_power * 0.5
+				glow.light_range = 0
+				glow.update_light()
+				continue
+			if(glow.light_color == "sw")
+				glow.light_color = "purple"
+				glow.light_power = glow.light_power * 2.27
+				glow.light_range = glow.range * 1.15
+				glow.update_light()
+				continue
+			if(glow.light_color == "purple")
+				glow.light_color = "se"
+				glow.light_power = glow.light_power * 0.44
+				glow.light_range = 0
+				glow.update_light()
+				continue
+			if(glow.light_color == "se")
+				glow.light_color = "#ffff00"
+				glow.light_range = glow.range * 0.9
+				glow.update_light()
+				continue
+			if(glow.light_color == "#ffff00")
+				glow.light_color = "ne"
+				glow.light_range = 0
+				glow.update_light()
+				continue
+			if(glow.light_color == "ne")
+				glow.light_color = "red"
+				glow.light_power = glow.light_power * 0.68
+				glow.light_range = glow.range * 0.85
+				glow.update_light()
+				continue
+		if(prob(2))
+			INVOKE_ASYNC(src, .proc/hierofunk)
+		sleep(selection.song_beat)
+
+
+/obj/machinery/disco/proc/dance(var/mob/living/carbon/M) //Show your moves
+	switch(rand(0,9))
+		if(0 to 1)
+			set waitfor = 0
+			for(var/i = 1, i < 8, i++)
+				M.SpinAnimation(7,1)
+				M.setDir(pick(cardinal))
+				sleep(10)
+		if(2 to 3)
+			set waitfor = 0
+			for(var/i in 1 to 6)
+				if (!M)
+					return
+				M.SpinAnimation(7,1)
+				M.setDir(pick(cardinal))
+				for (var/x in 1 to 12)
+					sleep(1)
+					if (!M)
+						return
+					if (i<5)
+						M.pixel_y += 1
+					if (i>4)
+						M.pixel_y -= 2
+					M.setDir(turn(M.dir, 90))
+					switch (M.dir)
+						if (NORTH)
+							M.pixel_y += 3
+						if (SOUTH)
+							M.pixel_y -= 3
+						if (EAST)
+							M.pixel_x -= 3
+						if (WEST)
+							M.pixel_x += 3
+				sleep(12)
+			M.pixel_x = 0
+			M.pixel_y = 0
+		if(4 to 6)
+			var/speed = rand(1,3)
+			set waitfor = 0
+			var/time = 30
+			while(time)
+				sleep(speed)
+				for(var/i in 1 to speed)
+					M.setDir(pick(cardinal))
+					M.lay_down(TRUE)
+				 time--
+
+		if(7 to 9)
+			M.setDir(get_dir(M, src))
+			spawn (0)
+				if (M)
+					animate(M, transform = matrix(180, MATRIX_ROTATE), time = 1, loop = 0)
+				sleep (70)
+				if (M)
+					animate(M, transform = null, time = 1, loop = 0)
+			for (var/i = 0, i < 60, i++)
+				if (!M)
+					return
+				if (i<31)
+					M.pixel_y += 1
+				if (i>30)
+					M.pixel_y -= 1
+				M.setDir(turn(M.dir, 90))
+				switch (M.dir)
+					if (NORTH)
+						M.pixel_y += 3
+					if (SOUTH)
+						M.pixel_y -= 3
+					if (EAST)
+						M.pixel_x -= 3
+					if (WEST)
+						M.pixel_x += 3
+				sleep (1)
+			M.pixel_x = 0
+			M.pixel_y = 0
```

Comment:
Again, why would you want the risk of permanently offset pixels, this is a crucial safety valve